TombEngine/Game/Lara/lara_helpers.cpp

```diff
@@ -68,28 +69,199 @@ void HandleLaraMovementParameters(ItemInfo* item, CollisionInfo* coll)
 		lara->Control.RunJumpQueued = false;
 
 	// Reset lean.
-	if ((!lara->Control.IsMoving || (lara->Control.IsMoving && !(TrInput & (IN_LEFT | IN_RIGHT)))) &&
+	if ((!lara->Control.IsMoving || (lara->Control.IsMoving && !(IsHeld(In::Left) || IsHeld(In::Right)))) &&
 		(!lara->Control.IsLow && item->Animation.ActiveState != LS_DEATH)) // HACK: Don't interfere with surface alignment in crouch, crawl, and death states.
 	{
 		ResetPlayerLean(item, 1 / 6.0f);
 	}
 
 	// Reset crawl flex.
 	if (!(TrInput & IN_LOOK) && coll->Setup.Height > LARA_HEIGHT - LARA_HEADROOM &&	// HACK
-		(!item->Animation.Velocity.z || (item->Animation.Velocity.z && !(TrInput & (IN_LEFT | IN_RIGHT)))))
+		(!item->Animation.Velocity.z || (item->Animation.Velocity.z && !(IsHeld(In::Left) || IsHeld(In::Right)))))
 	{
 		ResetPlayerFlex(item, 0.1f);
 	}
 
 	// Apply and reset turn rate.
 	item->Pose.Orientation.y += lara->Control.TurnRate;
-	if (!(TrInput & (IN_LEFT | IN_RIGHT)))
+	if (!(IsHeld(In::Left) || IsHeld(In::Right)))
 		lara->Control.TurnRate = 0;
 
 	lara->Control.IsLow = false;
 	lara->Control.IsMonkeySwinging = false;
 }
 
+static void UsePlayerMedipack(ItemInfo& item)
+{
+	auto& player = GetLaraInfo(item);
+
+	// Can't use medipack; return early.
+	if ((item.HitPoints <= 0 || item.HitPoints >= LARA_HEALTH_MAX) &&
```

Comment:
Does this condition potentially mean that you will be able to use medipack if you died of poisoning? I.e. health is 0, but poison is NOT 0?

Reply:
I will check this.

Reply:
Works ok, you can't heal if poisoned and dead.